### PR TITLE
Add src

### PR DESCRIPTION
**Pull request by yurybotov**
Add src and examples

### CODE REVIEW

File: examples/button/button.ino [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+void buttonHandler(bool state) {
+    if (state) { // if pressed
+        knob.setColor(255, 255, 255); // switch-on light
+    } else { // if not pressed
+        knob.setColor(0, 0, 0); // switch-off light
+    }
+}
+
+void setup() {
+    knob.begin(); // initialisation
```
Comment (AlexErAmp) [inline examples/button/button.ino]:
```suggestion
    knob.begin(); // initialization
```

File: examples/button/button.ino [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    knob.onButton(buttonHandler); // connect button handler
```
Comment (AlexErAmp) [inline examples/button/button.ino]:
```suggestion
    knob.onButton(buttonHandler); // attach a handler to the button click
```

File: examples/button/button.ino [hunk 3]
```diff
@@ -0,0 +1,29 @@
+/*
+* This example demonstrates pseudo-asynchronous operation with the RGBEncoder button.
+* The Knob lights up when the button is pressed.
+*/
+
+// Include library
+#include "RGBEncoder.h"
+
+// Create object of RGBEncoder
```
Comment (AlexErAmp) [inline examples/button/button.ino]:
```suggestion
// Create an RGBEncoder object
```

File: examples/colorsetter/colorsetter.ino [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+* The example demonstrates the operation of all RGBEncoder inner devices.
+* The encoder adjusts the RGB color components one by one. 
+* The button selects which color component is adjusted.
+* The LED displays the selected color.
+*/
+
+// Include library
+#include "RGBEncoder.h"
+
+// Create object of RGBEncoder
```
Comment (AlexErAmp) [inline examples/colorsetter/colorsetter.ino]:
```suggestion
// Create an RGBEncoder object
```

File: examples/colorsetter/colorsetter.ino [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+RGBEncoder knob;
+
+// mode is state mashine state.
```
Comment (AlexErAmp) [inline examples/colorsetter/colorsetter.ino]:
```suggestion
// mode is the state of a finite-state machine.
```

File: examples/colorsetter/colorsetter.ino [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+// By switching these states, we select the color component that we want to adjust.
+#define CHANGE_RED 0
+#define CHANGE_GREEN 1
+#define CHANGE_BLUE 2
+byte mode = CHANGE_RED;
+
+// Current color components values
+byte r = 0, g = 0, b = 0;
```
Comment (AlexErAmp) [inline examples/colorsetter/colorsetter.ino]: Use more descriptive names.

File: examples/colorsetter/colorsetter.ino [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        break;
+    case CHANGE_BLUE:
+        b = value;
+        break;
+    }
+    knob.setColor(r, g, b); // show new color
+}
+
+void setup() {
+    knob.begin(); // initialisation
```
Comment (AlexErAmp) [inline examples/colorsetter/colorsetter.ino]:
```suggestion
    knob.begin(); // initialization
```

File: examples/pulsegenerator/pulsegenerator.ino [hunk 8]
```diff
@@ -0,0 +1,46 @@
+/*
+ * This example demonstrates the operation of an RGBEncoder.
+ * This is a square wave generator with a frequency adjustable by an encoder in the range of 1...30 Hz.
+ */
+
+// Include library
+#include "RGBEncoder.h"
+
+// Create object of RGBEncoder
```
Comment (AlexErAmp) [inline examples/pulsegenerator/pulsegenerator.ino]: Create an RGBEncoder object

File: src/RGBEncoder.cpp [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+ * Product page: https://amperka.ru/product/rgbencoder
+ * © Amperka LLC (https://amperka.com, dev@amperka.com)
+ *
+ * Author: Yury Botov <by@amperka.com>
+ * License: GPLv3, all text here must be included in any redistribution.
+ */
+
+#include "RGBEncoder.h"
+
+void buttonDumb(bool state) { (void)state; }
```
Comment (AlexErAmp) [inline src/RGBEncoder.cpp]: These three functions can be declared `static`.

File: src/RGBEncoder.cpp [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+}
+
+void RGBEncoder::setStep(int32_t step) { _encoderStep = step; }
+
+void RGBEncoder::setMode(enum EncoderMode mode) { _encoderMode = mode; }
+
+void RGBEncoder::update() {
+    // button part
+    int16_t state = GpioExpander::digitalRead(_buttonPin);
+    if (state == 0 && _buttonState == true) {
```
Comment (AlexErAmp) [inline src/RGBEncoder.cpp]:
```suggestion
    if (state == LOW && _buttonState) {
```

File: src/RGBEncoder.cpp [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+        delay(_debounceTime);
+        state = GpioExpander::digitalRead(_buttonPin);
+        if (state == 0) {
```
Comment (AlexErAmp) [inline src/RGBEncoder.cpp]:
```suggestion
        if (state == LOW) {
```

File: src/RGBEncoder.cpp [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+            _buttonState = false;
+            _buttonHandler(false);
+        }
+    }
+    if (state != 0 && _buttonState == false) {
```
Comment (AlexErAmp) [inline src/RGBEncoder.cpp]:
```suggestion
    if (state == HIGH && !_buttonState) {
```

File: src/RGBEncoder.cpp [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+        delay(_debounceTime);
+        state = GpioExpander::digitalRead(_buttonPin);
+        if (state != 0) {
```
Comment (AlexErAmp) [inline src/RGBEncoder.cpp]:
```suggestion
        if (state == HIGH) {
```

**Review by AlexErAmp — CHANGES_REQUESTED** [review]

File: src/RGBEncoder.cpp (hunk 10 above)
Comment (yurybotov) [reply src/RGBEncoder.cpp]: need to test it. int16_t GpioExpander::digitalRead(_buttonPin) is very strange method

**Review by S-VIN — APPROVED** [review]
I don`t know, what to find fault with.

File: src/RGBEncoder.cpp (hunk 9 above)
Comment (yurybotov) [reply src/RGBEncoder.cpp]: Done

File: src/RGBEncoder.cpp (hunk 11 above)
Comment (yurybotov) [reply src/RGBEncoder.cpp]: Done

File: src/RGBEncoder.cpp (hunk 10 above)
Comment (yurybotov) [reply src/RGBEncoder.cpp]: Done

File: src/RGBEncoder.cpp (hunk 12 above)
Comment (yurybotov) [reply src/RGBEncoder.cpp]: Done

File: src/RGBEncoder.cpp (hunk 13 above)
Comment (yurybotov) [reply src/RGBEncoder.cpp]: Done

File: examples/colorsetter/colorsetter.ino (hunk 6 above)
Comment (yurybotov) [reply examples/colorsetter/colorsetter.ino]: Done

File: examples/pulsegenerator/pulsegenerator.ino (hunk 8 above)
Comment (yurybotov) [reply examples/pulsegenerator/pulsegenerator.ino]: Done

**Comment by yurybotov** [comment]
In addition to fixes in the code, API_ru.md was corrected and API.md was added

File: src/RGBEncoder.h [hunk 14]
```diff
@@ -0,0 +1,72 @@
+/*
+ * This file is a part of RGBEncoder library.
+ *
+ * Product page: https://amperka.ru/product/rgbencoder
```
Comment (nomad605dis) [inline src/RGBEncoder.h]:
```suggestion
 * Product page: https://amperka.ru/product/troyka-rgb-encoder
```

Comment (nomad605dis) [reply src/RGBEncoder.h]: And, I suggest remove all links to the product page.

**Review by nomad605dis — APPROVED** [review]
That's Ok, one small request in comment.

File: src/RGBEncoder.h (hunk 14 above)
Comment (yurybotov) [reply src/RGBEncoder.h]: Victor is unlikely to agree. This is a kind of ads.

File: src/RGBEncoder.cpp [hunk 15]
```diff
@@ -0,0 +1,141 @@
+/*
+ * This file is a part of RGBEncoder library.
+ *
+ * Product page: https://amperka.ru/product/rgbencoder
```
Comment (nomad605dis) [inline src/RGBEncoder.cpp]:
```suggestion
 * Product page: https://amperka.ru/product/troyka-rgb-encoder
```

File: API.md [hunk 16]
```diff
@@ -0,0 +1,90 @@
+# RGBEncoder API
+
+## `class RGBEncoder`
+
+Create an `RGBEncoder` object to work with the multifunctional module [RGBEncoder] (https://my.amperka.com/modules/RGBEncoder).
```
Comment (nomad605dis) [inline API.md]:
```suggestion
Create an `RGBEncoder` object to work with the multifunctional module [RGBEncoder] (https://my.amperka.com/modules/troyka-rgb-encoder).
```

File: API_ru.md [hunk 17]
```diff
@@ -81,4 +81,12 @@
 
 ### `void onEncoder(void (*encoderHandler)(int32_t value))`
 
-Регистрирует функцию-обработчик событий от энкодера.
+Регистрирует функцию-обработчик событий от энкодера. Функция обработчик вызывается в момент получения изменений и получает текущее значение. энкодера.
```
Comment (AlexErAmp) [inline API_ru.md]:
```suggestion
Регистрирует функцию-обработчик событий от энкодера. Функция-обработчик вызывается в момент получения изменений и получает текущее значение энкодера.
```

File: API_ru.md [hunk 18]
```diff
@@ -81,4 +81,12 @@
...
+
+### `void onStepLeft(void (*stepLeftHandler)())`
+
+Регистрирует функцию - обработчик событий от энкодера. Вызывается для каждого шага при вращении влево.
```
Comment (AlexErAmp) [inline API_ru.md]:
```suggestion
Регистрирует функцию-обработчик событий от энкодера. Вызывается для каждого шага при вращении влево.
```

File: API_ru.md [hunk 19]
```diff
@@ -81,4 +81,12 @@
...
+
+### `void onStepRight(void (*stepRightHandler)())`
+
+Регистрирует функцию - обработчик событий от энкодера. Вызывается для каждого шага при вращении вправо.
```
Comment (AlexErAmp) [inline API_ru.md]:
```suggestion
Регистрирует функцию-обработчик событий от энкодера. Вызывается для каждого шага при вращении вправо.
```

**Review by nomad605dis — COMMENTED** [review]
Ok, then two more minor edits.

File: API.md [hunk 20]
```diff
@@ -0,0 +1,90 @@
...
+This module is a rotation encoder, a button (when you press the encoder shaft), and RGB illumination of the encoder shaft. There are two interfaces for the button and encoder: direct access and pseudo-asynchronous.
+
+### `RGBEncoder(uint8_t i2cAddress = 42)`
+
+Creates a new `RGBEncoder` object using hardware i2c.
+
+- `i2cAddress`: i2c bus device address. When called without a parameter, its value is assumed to be 42.
+
+
```
Comment (AlexErAmp) [inline API.md]: An unnecessary empty string. Please run through prettier (API_ru.md too).

File: API.md [hunk 21]
```diff
@@ -0,0 +1,90 @@
...
+Sets the step of changing the encoder value.
+
+- `step`: step of changing the encoder value.
+
+### `void setMode(enum EncoderMode mode)`
+
+Sets the operation mode of the encoder - with clamp on the range or cyclic.
+
+- `mode`: the set mode of the encoder, can be` RGBENC_CLAMP` or `RGBENC_WRAP`.
```
Comment (AlexErAmp) [inline API.md]:
```suggestion
- `mode`: the set mode of the encoder, can be `RGBENC_CLAMP` or `RGBENC_WRAP`.
```

File: API.md [hunk 22]
```diff
@@ -0,0 +1,90 @@
...
+
+### `void update()`
+
+Synchronizes the encoder state. Call it in `loop ()`.
```
Comment (AlexErAmp) [inline API.md]:
```suggestion
Synchronizes the encoder state. Call it in `loop()`.
```

**Review by AlexErAmp — CHANGES_REQUESTED** [review]
Small fixes.

File: src/RGBEncoder.cpp (hunk 15 above)
Comment (yurybotov) [reply src/RGBEncoder.cpp]: ok

File: API.md (hunk 16 above)
Comment (yurybotov) [reply API.md]: ok

File: API_ru.md (hunk 17 above)
Comment (yurybotov) [reply API_ru.md]: ok

File: API_ru.md (hunk 18 above)
Comment (yurybotov) [reply API_ru.md]: ok

File: API_ru.md (hunk 19 above)
Comment (yurybotov) [reply API_ru.md]: ok

File: API.md (hunk 21 above)
Comment (yurybotov) [reply API.md]: ok

File: API.md (hunk 22 above)
Comment (yurybotov) [reply API.md]: ok

File: API.md (hunk 20 above)
Comment (yurybotov) [reply API.md]: Done

**Review by AlexErAmp — APPROVED** [review]
LGTM

File: API.md [hunk 23]
```diff
@@ -0,0 +1,90 @@
+# RGBEncoder API
+
+## `class RGBEncoder`
+
+Create an `RGBEncoder` object to work with the multifunctional module [RGBEncoder] (https://my.amperka.com/modules/troyka-rgb-encoder).
+This module is a rotation encoder, a button (when you press the encoder shaft), and RGB illumination of the encoder shaft. There are two interfaces for the button and encoder: direct access and pseudo-asynchronous.
```
Comment (Matthew3000) [inline API.md]:
```suggestion
This module is a rotation encoder, a button (press the encoder shaft), and RGB illumination of the encoder shaft. There are two interfaces for the button and encoder: direct access and pseudo-asynchronous.
```

File: API.md [hunk 24]
```diff
@@ -0,0 +1,90 @@
...
+### `void begin(TwoWire* wire)`
+
+Initializes the interface, sets the device to the "default" state. In this case, the operating range of the encoder values ​​is set [0-255], the value of the encoder at start is 0, the step of change is 1, the mode is "within the range".
+Call this method before interacting with RGBEncoder. For example, in the `setup ()` function.
+
+- `wire`: selected i2c interface. When called without a parameter, the `Wire` hardware interface.
+
+### `void changeAddr(uint8_t newAddress)`
+
+- `newAddress`: new i2c address for device.
```
Comment (Matthew3000) [inline API.md]:
```suggestion
- `newAddress`: new i2c address for the device.
```

File: API.md [hunk 25]
```diff
@@ -0,0 +1,90 @@
...
+
+### `void saveAddr(void)`
+
+Saves the current device i2c address to EEPROM.
+
+### `void setValue(int32_t value)`
+
+Forces the current value of the encoder. If this value is outside the working range, it is converted to the extreme value of the working range.
```
Comment (Matthew3000) [inline API.md]:
```suggestion
Forces the current value of the encoder. If this value is outside of the working range, it is converted to the working range's extreme value.
```

File: API.md [hunk 26]
```diff
@@ -0,0 +1,90 @@
...
+
+- `value`: the set value of the encoder [−2147483648; 2147483647] must be within the range set by the `setRange` command.
+
+### `void setRange(int32_t min, int32_t max)`
+
+Sets the limits of the working range of the encoder values. If the current value is outside the operating range, it is converted to the extreme value of the operating range.
```
Comment (Matthew3000) [inline API.md]:
```suggestion
Sets the limits of the working range of the encoder values. If the current value is outside of the operating range, it is converted to the operating range's extreme value.
```

File: API.md [hunk 27]
```diff
@@ -0,0 +1,90 @@
...
+
+- `min`: lower limit of the encoder working range [−2147483648; 2147483647].
+- `max`: upper limit of the encoder working range [−2147483648; 2147483647].
+
+### `void setStep(int32_t step)`
+
+Sets the step of changing the encoder value.
```
Comment (Matthew3000) [inline API.md]:
```suggestion
Sets the step of change of the encoder value.
```

File: API.md [hunk 28]
```diff
@@ -0,0 +1,90 @@
...
+
+- `step`: step of changing the encoder value.
```
Comment (Matthew3000) [inline API.md]:
```suggestion
- `step`: the step of change of the encoder value.
```

File: API.md [hunk 29]
```diff
@@ -0,0 +1,90 @@
...
+
+Returns the current debounced state of the button.
+
+### `int32_t getEncoderValue() const`
+
+Returns the current value of the encoder.
+
+### `void onButton(void (*buttonHandler)(bool state))`
+
+Registers a function - an event handler from a button.
```
Comment (Matthew3000) [inline API.md]:
```suggestion
Registers a function - a button event handler.
```

**Review by Matthew3000 — CHANGES_REQUESTED** [review]

File: API.md (hunk 23 above)
Comment (yurybotov) [reply API.md]: Ok

File: API.md (hunk 24 above)
Comment (yurybotov) [reply API.md]: Ok

File: API.md (hunk 25 above)
Comment (yurybotov) [reply API.md]: Ok

File: API.md (hunk 26 above)
Comment (yurybotov) [reply API.md]: Ok

File: API.md (hunk 27 above)
Comment (yurybotov) [reply API.md]: Ok

File: API.md (hunk 28 above)
Comment (yurybotov) [reply API.md]: Ok

File: API.md (hunk 29 above)
Comment (yurybotov) [reply API.md]: Ok

**Review by Matthew3000 — APPROVED** [review]

File: examples/button/button.ino [hunk 30]
```diff
@@ -0,0 +1,29 @@
...
+RGBEncoder knob;
+
+// Button handler
+void buttonHandler(bool state) {
+    if (state) { // if pressed
+        knob.setColor(255, 255, 255); // switch-on light
+    } else { // if not pressed
+        knob.setColor(0, 0, 0); // switch-off light
+    }
+}
```
Comment (nkrkv) [inline examples/button/button.ino]: Can be more self-documenting:

```suggestion
void buttonHandler(bool pressed) {
    if (pressed) {
        knob.setColor(255, 255, 255); // switch-on light
    } else {
        knob.setColor(0, 0, 0); // switch-off light
    }
}
```

File: examples/button/button.ino [hunk 31]
```diff
@@ -0,0 +1,29 @@
...
+
+void setup() {
+    knob.begin(); // initialization
```
Comment (nkrkv) [inline examples/button/button.ino]:
```suggestion
    knob.begin(); // initialize
```

File: examples/button/button.ino [hunk 32]
```diff
@@ -0,0 +1,29 @@
...
+    knob.onButton(buttonHandler); // attach a handler to the button click
+    knob.setColor(0, 0, 0); // switch-off light
+}
+
+void loop() {
+    knob.update(); // it implements pseudo-asynchronous work
```
Comment (nkrkv) [inline examples/button/button.ino]:
```suggestion
    knob.update(); // upkeep to trigger the handler when necessary
```

File: examples/colorsetter/colorsetter.ino [hunk 33]
```diff
@@ -0,0 +1,74 @@
...
+
+// Create an RGBEncoder object
+RGBEncoder knob;
+
+// mode is the state of a finite-state machine.
+// By switching these states, we select the color component that we want to adjust.
+#define CHANGE_RED 0
+#define CHANGE_GREEN 1
+#define CHANGE_BLUE 2
+byte mode = CHANGE_RED;
```
Comment (nkrkv) [inline examples/colorsetter/colorsetter.ino]: C’mon. 2020 is here. Use enums and/or constexprs. `#define`’s are never required for constants nowadays.

File: examples/colorsetter/colorsetter.ino [hunk 34]
```diff
@@ -0,0 +1,74 @@
...
+
+// Current color components values
+byte red = 0, green = 0, blue = 0;
+
+// Handler of button
+void buttonHandler(bool state) {
+    if (state) { // if button pressed ...
```
Comment (nkrkv) [inline examples/colorsetter/colorsetter.ino]: state -> pressed.

Prefer early exit to remove nesting:

```cpp
void buttonHandler(bool pressed) {
    if (!pressed)
        return;

    switch (mode) {
    // ...
```

File: examples/colorsetter/colorsetter.ino [hunk 35]
```diff
@@ -0,0 +1,74 @@
...
+    knob.begin(); // initialization
+    // register handlers for button and encoder
+    knob.onButton(buttonHandler);
+    knob.onEncoder(encoderHandler);
+    // switch-of led
+    knob.setColor(0, 0, 0);
+}
+
+void loop() {
+    knob.update(); // it implements pseudo-asynchronous work
```
Comment (nkrkv) [inline examples/colorsetter/colorsetter.ino]: ditto

File: examples/colorsetter/colorsetter.ino [hunk 36]
```diff
@@ -0,0 +1,74 @@
...
+    }
+    knob.setColor(red, green, blue); // show new color
+}
+
+void setup() {
+    knob.begin(); // initialization
+    // register handlers for button and encoder
+    knob.onButton(buttonHandler);
+    knob.onEncoder(encoderHandler);
+    // switch-of led
```
Comment (nkrkv) [inline examples/colorsetter/colorsetter.ino]:
```suggestion
    // switch-off led
```

File: src/RGBEncoder.h [hunk 37]
```diff
@@ -0,0 +1,72 @@
...
+    static constexpr uint8_t _buttonPin = 3;
+    static constexpr uint16_t _debounceTime = 20; // ms
+
+    uint8_t _i2cAddress;
+    bool _buttonState;
+    int32_t _encoderLowerBound;
+    int32_t _encoderUpperBound;
+    int32_t _encoderCurrentValue;
+    uint32_t _encoderStep;
+    enum EncoderMode _encoderMode;
```
Comment (nkrkv) [inline src/RGBEncoder.h]:
```suggestion
    EncoderMode _encoderMode;
```

C++ is here, not C

File: library.properties [hunk 38]
```diff
@@ -0,0 +1,9 @@
+name=Amperka RGBEncoder
+version=0.1.0
+author=Amperka <dev@amperka.com>
+maintainer=Amperka <amperka.ru>
+sentence=A library to interface with the RGBEncoder.
+paragraph=A library to interface with the RGBEncoder.
```
Comment (nkrkv) [inline library.properties]: Why duplicated?

File: library.properties [hunk 39]
```diff
@@ -0,0 +1,9 @@
+name=Amperka RGBEncoder
+version=0.1.0
```
Comment (nkrkv) [inline library.properties]:
```suggestion
version=1.0.0
```

Let’s start with a confident 1.0.0 guaranteeing backward compatibility within 1.x.x for customers

**Review by nkrkv — CHANGES_REQUESTED** [review]

File: library.properties (hunk 39 above)
Comment (yurybotov) [reply library.properties]: OK

File: examples/colorsetter/colorsetter.ino (hunk 36 above)
Comment (yurybotov) [reply examples/colorsetter/colorsetter.ino]: OK

File: examples/button/button.ino (hunk 32 above)
Comment (yurybotov) [reply examples/button/button.ino]: OK

File: examples/colorsetter/colorsetter.ino (hunk 35 above)
Comment (yurybotov) [reply examples/colorsetter/colorsetter.ino]: Done

File: examples/button/button.ino (hunk 31 above)
Comment (yurybotov) [reply examples/button/button.ino]: OK

File: examples/button/button.ino (hunk 30 above)
Comment (yurybotov) [reply examples/button/button.ino]: Ok

File: examples/colorsetter/colorsetter.ino (hunk 33 above)
Comment (yurybotov) [reply examples/colorsetter/colorsetter.ino]: Ok

File: examples/colorsetter/colorsetter.ino (hunk 34 above)
Comment (yurybotov) [reply examples/colorsetter/colorsetter.ino]: Done

File: src/RGBEncoder.h (hunk 37 above)
Comment (yurybotov) [reply src/RGBEncoder.h]: Ok

File: library.properties (hunk 38 above)
Comment (yurybotov) [reply library.properties]: Done

File: src/RGBEncoder.h [hunk 40]
```diff
@@ -0,0 +1,72 @@
...
+    void begin();
+    void begin(TwoWire* wire);
+
+    void changeAddr(uint8_t newAddress);
+    void saveAddr();
+
+    void setValue(int32_t value);
+    void setRange(int32_t min, int32_t max);
+    void setStep(int32_t step);
+    void setMode(enum EncoderMode mode);
```
Comment (nkrkv) [inline src/RGBEncoder.h]:
```suggestion
    void setMode(EncoderMode mode);
```

File: API_ru.md [hunk 41]
```diff
@@ -34,38 +34,38 @@
 
 Принудительно устанавливает текущее значение энкодера. Если это значение выходит за пределы рабочего диапазона оно преобразуется в крайнее значение рабочего диапазона.
 
-- `value`: устанавливаемое значение энкодера [−2147483648;2147483647] должно быть в пределах диапазона установленного командой `setRange`. 
+- `value`: устанавливаемое значение энкодера [−2147483648;2147483647] должно быть в пределах диапазона установленного командой `setRange`.
```
Comment (nkrkv) [inline API_ru.md]:
```suggestion
- `value`: устанавливаемое значение энкодера [−2147483648; 2147483647] должно быть в пределах диапазона установленного командой `setRange`.
```

File: API_ru.md [hunk 42]
```diff
@@ -34,38 +34,38 @@
...
 
 ### `void setRange(int32_t min, int32_t max)`
 
 Задает границы рабочего диапазона значений энкодера. Если текущее значение выходит за пределы рабочего диапазона оно преобразуется в крайнее значение рабочего диапазона.
 
-- `min`: нижняя граница рабочего диапазона энкодера [−2147483648;2147483647]. 
-- `max`: верхняя граница рабочего диапазона энкодера [−2147483648;2147483647]. 
+- `min`: нижняя граница рабочего диапазона энкодера [−2147483648;2147483647].
+- `max`: верхняя граница рабочего диапазона энкодера [−2147483648;2147483647].
```
Comment (nkrkv) [inline API_ru.md]:
```suggestion
- `min`: нижняя граница рабочего диапазона энкодера [−2147483648; 2147483647].
- `max`: верхняя граница рабочего диапазона энкодера [−2147483648; 2147483647].
```

**Review by nkrkv — APPROVED** [review]

File: API_ru.md (hunk 42 above)
Comment (yurybotov) [reply API_ru.md]: Ok

File: API_ru.md (hunk 41 above)
Comment (yurybotov) [reply API_ru.md]: Ok

File: src/RGBEncoder.h (hunk 40 above)
Comment (yurybotov) [reply src/RGBEncoder.h]: Ok

**Comment by yurybotov** [comment]
Ufffffff

**Pull request merged**